R/getSettings.R

```diff
@@ -1,66 +1,60 @@
-} else if (file.exists(paste0("/mnt///c/Users/", Sys.getenv("USER"), "/AppData/Roaming/Code/User/settings.json"))) {
-    # WSL
-    file(paste0("/mnt///c/Users/", Sys.getenv("USER"), "/AppData/Roaming/Code/User/settings.json"))
+homedir <- Sys.getenv(
+    if (.Platform$OS.type == "windows") "USERPROFILE" else "HOME"
+)
+
+fileCon <- if (file.exists(file.path(homedir, ".vscode-R",  "settings.json"))) {
+    file(file.path(homedir, ".vscode-R", "settings.json"))
 } else {
     NULL
 }
 
-# settings.json can result in errors if read via fromJSON
-vsc_settings <- suppressWarnings(jsonlite::fromJSON(paste(readLines(fileCon), collapse = "")))
-ops <- Reduce(c, vsc_settings[grep("r.rOptions", names(vsc_settings))])
+if (!is.null(fileCon)) {
+    # settings.json can result in errors if read via fromJSON
+    vsc_settings <- suppressWarnings(
+        jsonlite::fromJSON(paste(readLines(fileCon), collapse = ""))[["rOptions"]]
+    )
+    ops <- Reduce(c, vsc_settings)
 
-# non-string values have to be converted from
-# strings due to VSC settings limitations
-get_val <- function(x) {
-    if (is.logical(x)) {
-        x
-    } else {
-        switch(EXPR = x,
-            "Two" = "Two",
-            "Active" = "Active",
-            "Beside" = "Beside",
-            "FALSE" = FALSE,
-            "TRUE" = TRUE,
-            "0" = 0,
-            "2" = 2,
+    # non-string values have to be converted from
+    # strings due to VSC settings limitations
+    get_val <- function(x) {
+        if (is.logical(x)) {
             x
-        )
+        } else {
+            switch(EXPR = x,
+                "Two" = "Two",
+                "Active" = "Active",
+                "Beside" = "Beside",
+                "FALSE" = FALSE,
+                "TRUE" = TRUE,
+                "0" = 0,
+                "2" = 2,
+                x
+            )
+        }
     }
-}
 
+    lapply(names(ops), function(x) {
+        val <- get_val(ops[[x]])
 
-lapply(names(ops), function(x) {
-    val <- get_val(ops[[x]])
-
-    # lhs = vscode setting name
-    # rhs name = R options name
-    switch(EXPR = x,
-        # arrays
-        "vsc.plot" = options("vsc.plot" = val),
-        "vsc.browser" = options("vsc.browser" = val),
-        "vsc.viewer" = options("vsc.viewer" = val),
-        "vsc.pageViewer" = options("vsc.page_viewer" = val),
-        "vsc.view" = options("vsc.view" = val),
-        "vsc.helpPanel" = options("vsc.helpPanel" = val),
-        "vsc.strMaxLevel" = options("vsc.str.max.level" = val),
-        # bools
-        "vsc.rstudioapi" = options("vsc.rstudioapi" = val),
-        "vsc.useHttpgd" = options("vsc.use_httpgd" = val),
-        "vsc.globalEnv" = options("vsc.globalenv" = val),
-        "vsc.showObjectSize" = options("vsc.show_object_size" = val)
-    )
-})
+        # lhs = vscode setting name
+        # rhs name = R options name
+        switch(EXPR = x,
+            # arrays
+            "vsc.plot" = options("vsc.plot" = val),
```

Comment:
Such camelCase to snake_case translation looks quite redundant and confusing. Would it be helpful if we keep the original names in R?

Reply:
I agree - the names were mostly a placeholder for the time being. I think ideally the names would be more descriptive, so that someone that hasn't come across the wiki before could understand what they do

Reply:
More broadly, are there any opinions/ideas for how these options should:

1. be organised? At the moment it's under a fairly non-descriptive "bools" and "arrays", but this clumps together options that may only be tangentially related. Also, would these options better fall under already-defined categories, such as `r.rterm.rOptions`?

2. be named? One idea I had was have descriptive names, and then in the description state the option's name. E.g., 

**Show Object Size** (showObjectSize)
Show object size when hovering over a workspace viewer item.
Sets the option `vsc.show_object_size` in R.

Reply:
Re: categorising the settings, another option would be to split these under already-existing related categories. E.g., plot options all moved to `r.plots`, viewer options moved under session, etc.

Reply:
Now that all these options are only useful for vscode-R, there might be a point to only keep them in vscode settings and deprecate these options in R. Then it is much easier to put each setting in the appropriate category.

Reply:
I agree going forward that it's better to keep this on the VSCode side, but  for backwards compatibility it might be a good idea to keep the options around and deprecate them a bit further down the line

Reply:
> I agree going forward that it's better to keep this on the VSCode side, but for backwards compatibility it might be a good idea to keep the options around and deprecate them a bit further down the line

Yes, exactly. Just go ahead with the most consistent names and categories in vscode. Maybe it is useful to put the corresponding R option in the settings description to make it easier to search.